src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
```

Comment:
We can optimize this solution by avoiding double enumeration. Create an object with `mother` as a key before `women` variable definition
```
const peopleObject = people.reduce((acc, p) => ({...acc, [p.mother]: p}),{})
```
and then we can replace `people.find((p1) => p1.mother === p.name)` by `peopleObject[p.name]`